\subsection{Datasets}

We used the same synthetic dataset from Loveâ€™s 2018 method article on DTU analysis, 
which consists of 24 sets of unstranded inward pair-ended reads \cite{love_swimming_2018}. 
This data was generated using polyester 1.16.0 and alpine version 1.6.0. \cite{frazee2015polyester} 
We also utilized the Homo sapiens GRCh37 version 28 sequence as our 
human transcriptome reference, which was the same reference used for the generation 
of the simulated read. \cite{noauthor_homo_sapiens_nodate}

% We used synthetic datasets in the same form as \cite{patro_salmon_2017}. 
% We generated our synthetic data using polyester 1.16.0 and alpine version 1.6.0. 
% \cite{frazee2015polyester}\cite{love2016modeling} There was ground truth available for this dataset. We additionally used real human datasets. 
% We used a dataset consisting of NK cells, T cells, and tumor from matched soft tissue sarcoma and peripheral 
% blood collected from soft tissue sarcomas patients who had undergone surgery. 
% \cite{judge_transcriptome_2022} We also utilized the human transcriptome reference: Homo sapiens GRCh38 cDNA sequence. 
% These were gene models built from alignments of the human proteome and alignments of human cDNAs. \cite{noauthor_homo_sapiens_nodate}
